### ISSUE TITLE
Output names do not match input code.

### ISSUE THREAD

**Issue by Lumengine** [shader-slang/slang#1789]
An simple input shader like this: 
```
struct AssembledVertex 
{ 
    float3	position : POSITION; 
    float3	color    : COLOR; 
};

struct CoarseVertex
{
    float3 color;
};
// Vertex  Shader

struct VertexStageOutput
{
    CoarseVertex    coarseVertex    : CoarseVertex;
    float4          sv_position     : SV_Position;
};
[shader("vertex")]
VertexStageOutput vertexMain(
    AssembledVertex assembledVertex)
{
    VertexStageOutput output;

    float3 position = assembledVertex.position;
    float3 color    = assembledVertex.color;

    output.coarseVertex.color = color;
    output.sv_position = float4(position, 1.0);

    return output;
}
```
will generate SPIRV binaries with modified input names : _S2 and _S3 for this particular case instead of `position, color`
In case of SPIRV reflection (with spirv cross for example), it could be annoying. T. Foley suggested me to open this issue to maybe add this behaviour as an option.
I'm pretty new to Slang, so I may have missed something.
From what I've seen, it's a great API. Thanks for everything !

**Comment by Lumengine** [comment]
Up ?

**Comment by tfoleyNV** [comment]
Hi. Sorry this kind of slid off our priority list. If you give us some idea of your timeline for when you’d need this feature I can try to make sure this gets some attention.

The one thing to note is that the preferred model for using Slang when you require reflection information is to use the reflection API Slang provides. The exact symbol names in output GLSL (and hence SPIR-V) are supposed to just be an implementation detail. We can try to support this kind of use case in a best-effort way, but it will always be a bit off of the code path that our main users rely on.

**Comment by Lumengine** [comment]
Thanks for the reply. I tried the slang reflection API, and have question about it. I don't know if I can ask them here or not but let's try: if (after specialization of all interfaces etc...) I want to loop over shader resources and get the name, desciptor set, binding (I'm working with Vulkan), type and memory accesses, what is the preferred way to do so ? I tried to look at your render-vk.cpp but it's a bit hard to follow all the code tbh.
